pkg/dogstatsd/batch.go

```diff
@@ -18,12 +18,15 @@ type batcher struct {
 	choutSamples       chan<- []metrics.MetricSample
 	choutEvents        chan<- []*metrics.Event
 	choutServiceChecks chan<- []*metrics.ServiceCheck
+
+	metricSamplePool *metrics.MetricSamplePool
 }
 
 func newBatcher(aggregator *aggregator.BufferedAggregator) *batcher {
```

Comment:
It'd be nice to avoid shadowing the `aggregator` package with the `aggregator` variable, i.e.:

```suggestion
func newBatcher(agg *aggregator.BufferedAggregator) *batcher {
```

wdyt?

(note: I'm surprised one of our linters doesn't check for this, but this might be common in Go code...)